### PR TITLE
ES-2153 - Added support for multiple files and for updating default parameters

### PR DESCRIPTION
**Pull request by Andrei-Predoiu**
## Added support for multiple files. 
* If `directory` is set to a folder, dependabot-circleci will travel the whole directory tree and check all yml/yaml files.
* Default for `directory` is now `.circleci/config.yml`. **This different than the old behavior and will break stuff in our templates.**
## Added support for updating the default value of parameters 
* **!Only works if the image or orb is defined in the same file as the parameter!**
* If a default is not set or a parameter is not found then nothing will be updated

Example: 
https://github.com/BESTSELLER/cci-common-test/pulls
Files that can't be updated:
https://github.com/BESTSELLER/cci-common-test/blob/49ca40c5737faf0750552dfeeb7b53c58ae8e51d/src/jobs/build_n_push_docker.yml#L11

### CODE REVIEW

**Comment by Andrei-Predoiu** [comment]
@Gaardsholt do you think there will be timeout issues because of how long it can take to run this now?

**Review by Gaardsholt — COMMENTED** [review]
Can you please write a proper title for the pull request, just to make the release notes a bit better

I'm not completely done reviewing this pull request, I will continue when you have addressed the comments I have added so far :)

File: README.md [hunk 1]
```diff
@@ -9,20 +9,25 @@
 <br/>
 
 `dependabot-circleci` is, as its name suggests, a small dependabot for CircleCI orbs and container images.
-We have created this as at the time of creation it was nearly impossible to get changes into the official [dependabot](https://github.com/dependabot/dependabot-core).
+We have created this as at the time of creation it was nearly impossible to get changes into the
+official [dependabot](https://github.com/dependabot/dependabot-core).
```
[the remark this answers is not in the document]
Comment (Andrei-Predoiu) [reply README.md:13]: The linter on my IDE did it.

File: circleci/orb.go [hunk 2]
```diff
@@ -12,31 +12,44 @@ import (
...
-func extractOrbs(orbs []*yaml.Node) map[string]*yaml.Node {
+func extractOrbs(orbs []*yaml.Node, parameters *map[string]*yaml.Node) map[string]*yaml.Node {
 	updates := map[string]*yaml.Node{}
 	for i := 0; i < len(orbs); i = i + 2 {
 		orb := orbs[i+1]
 
 		log.Debug().Msg(fmt.Sprintf("current orb version: %s", orb.Value))
-		orbVersion := findNewestOrbVersion(orb.Value)
-		log.Debug().Msg(fmt.Sprintf("new orb version: %s", orbVersion))
+		orbRoot, currentVer, newestBer := findNewestOrbVersion(orb.Value, parameters)
```
Comment (dennisc87) [inline circleci/orb.go:21]: Should the parameter name have been `newestVer`?

File: circleci/update_test.go [hunk 3]
```diff
@@ -28,24 +27,22 @@ func getTestCases() map[string]*yaml.Node {
...
+	SHA := "ABC"
+	updates := map[string]Update{}
 
 	for k, v := range tests {
 		t.Run(k, func(t *testing.T) {
-			GetUpdates(v)
-			// if got := GetUpdates(v); !reflect.DeepEqual(got, tt.want) {
-			// 	t.Errorf("GetUpdates() = %v, want %v", got, tt.want)
+			ScanFileUpdates(&updates, v, &k, &SHA)
+			// if got := ScanFileUpdates(v); !reflect.DeepEqual(got, tt.want) {
```
Comment (dennisc87) [inline circleci/update_test.go:44]: Remove the outcommented code? Or leave a comment about why its still there?

**Review by Gaardsholt — APPROVED** [review]
LGTM - kinda hard to follow the changes, but I trust you :)

Could we add test cases for this new functionality?

**Review by Gaardsholt — APPROVED** [review]
